cmd/query/app/additional_headers_handler.go

```diff
@@ -5,13 +5,15 @@ package app
 
 import (
 	"net/http"
+
+	"go.opentelemetry.io/collector/config/configopaque"
 )
 
-func additionalHeadersHandler(h http.Handler, additionalHeaders http.Header) http.Handler {
+func additionalHeadersHandler(h http.Handler, additionalHeaders map[string]configopaque.String) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
if you use OTEL's ToServer then it already implements a handler for response handlers

Reply:
@yurishkuro is that something we want to do in this PR? I was thinking we could keep of this PR to just consolidating the configurations and then in the next one I can attempt to make this change. Let me know what you think.